src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java

```diff
@@ -622,4 +598,46 @@ private Properties initializePropertiesAndCheckFormat(String format, Map<String,
         }
         return exportProperties;
     }
+
+    private ExportNetworkInfos getExportNetworkInfos(Network network, String format,
+                                                    String fileOrNetworkName, Properties exportProperties,
+                                                    long networkSize) throws IOException {
+        Path tempDir = Files.createTempDirectory("export_", PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------")));
+        try {
+            DirectoryDataSource dataSource = new DirectoryDataSource(tempDir, fileOrNetworkName);
+            network.write(format, exportProperties, dataSource);
+
+            Set<String> fileNames = dataSource.listNames(".*");
+            if (fileNames.isEmpty()) {
+                throw new IOException("No files were created during export");
+            }
+
+            if (fileNames.size() == 1) {
+                Path fileDir = tempDir.resolve(fileNames.iterator().next());
+                return new ExportNetworkInfos(fileDir.getFileName().toString(), fileDir, networkSize);
+            }
+
+            Path zipFile = createZipFile(tempDir, fileOrNetworkName, fileNames);
+            return new ExportNetworkInfos(zipFile.getFileName().toString(), zipFile, networkSize);
+        } catch (IOException e) {
+            throw NetworkConversionException.failedToStreamNetworkToFile(e);
+        }
+    }
+
+    private Path createZipFile(Path tempDir, String fileOrNetworkName, Set<String> fileNames) throws IOException {
+        Path zipFile = tempDir.resolve(fileOrNetworkName + ".zip");
+        try (ZipOutputStream zos = new ZipOutputStream(Files.newOutputStream(zipFile))) {
+            for (String fileName : fileNames) {
+                Path sourceFile = tempDir.resolve(fileName);
+                if (Files.exists(sourceFile)) {
```

Comment:
I think you can remove this control. You just get the filenames few code rows before